### ISSUE TITLE
JPEG image is "blocky"

### ISSUE THREAD

**Issue by oliver-moran** [eugeneware/jpeg-js#17]
First, thanks very much for your library, which I am using in [Jimp](https://www.npmjs.com/package/jimp). Without this library and `pngjs` Jimp would not be possible.

An issue was raised about the picture below being "blocky" when it is decoded (especially the blurry background elements): https://github.com/oliver-moran/jimp/issues/134

A test case using Jimp is as follows:

``` js
var Jimp = require("jimp");

Jimp.read("image.jpg").then(image => {
    image.quality(100).write("out.png");
});
```

If I convert the image to a PNG using GIMP and perform the same test on that image then the issue doesn't present. For that reason I think the issue lies here.

![image](https://cloud.githubusercontent.com/assets/414918/14903781/76a1280c-0d9b-11e6-9b4d-33a94cea60af.jpg)

**Comment by ArmorDarks** [comment]
Hm, seems like library no longer maintained...

**Comment by benwiley4000** [comment]
Thanks @oliver-moran, and sorry for the delayed response. Could you confirm this issue still exists after recent updates?

**Comment by benwiley4000** [comment]
Cool!

**Comment by hipstersmoothie** [comment]
IDK if it's fully fixed though. I found another issue where encoding and decoding right after one another produces a slightly different image.

```js
const fs = require('fs')
const jpeg = require('jpeg-js');
const jpegData = fs.readFileSync('./exampleImages/npm.jpg');
const rawImageData = jpeg.decode(jpegData);
const jpegImageData = jpeg.encode(rawImageData, 100);

const Jimp = require('jimp')

fs.writeFileSync('./out.jpg', jpegImageData.data)
```

![npm](https://user-images.githubusercontent.com/1192452/43351718-0dd05e7c-91cb-11e8-85f0-f4ab16810133.jpg)

![out](https://user-images.githubusercontent.com/1192452/43351728-2f5738d6-91cb-11e8-9e70-55fd51c2dab4.jpg)

**Comment by benwiley4000** [comment]
Ok I'll reopen

**Comment by hipstersmoothie** [comment]
It almost looks like its grouping together 4 pixels and only using 1 color from the 4

**Comment by patrickhulce** [comment]
The original issue may have been related to and partially fixed by #25 if they were progressive JPEGs. If not, probably separate issue we should try to hunt down :)

Given https://github.com/oliver-moran/jimp/issues/121, it seems like it's still definitely on the decoding side not encoding side?

**Comment by hipstersmoothie** [comment]
yeah that makes sense since it happens when just deciding and then encoding to another format
